src/Solvers/batched_tridiagonal_solver.jl

```diff
@@ -1,26 +1,26 @@
 using Oceananigans.Architectures: arch_array
-
 import Oceananigans.Architectures: architecture
 
 """
     BatchedTridiagonalSolver
 
 A batched solver for large numbers of triadiagonal systems.
 """
-struct BatchedTridiagonalSolver{A, B, C, T, G, P}
-               a :: A
-               b :: B
-               c :: C
-               t :: T
-            grid :: G
-      parameters :: P
+struct BatchedTridiagonalSolver{A, B, C, T, G, P, D}
+                        a :: A
+                        b :: B
+                        c :: C
+                        t :: T
+                     grid :: G
+               parameters :: P
+    tridiagonal_direction :: D
```

Comment:
What do we think about the name of this property? Would `tridiagonal_dimension` be better or is `tridiagonal_direction` best?

Reply:
I prefer `tridiagonal_direction` since in English we generally tend to say `x`-direction and so on, instead of `x`-dimension. Plus I think mist of the docs use the words direction. For example from the `RectilinearGrid` docstring:

https://github.com/CliMA/Oceananigans.jl/blob/e0f0942de9f009944b21bb2ec480fe94573a4e33/src/Grids/rectilinear_grid.jl#L89-L100

Reply:
mm yes the docs use "direction". So we keep the same terminology. That's a good reason.

I was also reflecting on using `::Val{:x}` rather than `XDirection`, etc. We already have a `ZDirection` type right? Should we use `tridiagonal_direction=ZDirection()` rather than `tridiagonal_direction=:z`?